remix/app1/webpack.server.js

```diff
@@ -0,0 +1,171 @@
+import * as fs from "node:fs";
+import * as path from "node:path";
+
+import {readConfig} from "@remix-run/dev/dist/config.js";
+import {EsbuildPlugin} from "esbuild-loader";
+import nodeExternals from "webpack-node-externals";
+import {default as Enhanced} from '@module-federation/enhanced'
+import {default as NFP} from '@module-federation/node'
+
+const {AsyncBoundaryPlugin, ModuleFederationPlugin} = Enhanced
+const {UniversalFederationPlugin} = NFP
+import {getManifest} from "./utils/manifest.js";
+
+const mode =
+  process.env.NODE_ENV === "production" ? "production" : "development";
+const remixConfig = await readConfig();
+const isModule = remixConfig.serverModuleFormat === "esm";
+
+console.log({isModule});
+
+if (!isModule) {
+  fs.writeFileSync('./build/package.json', JSON.stringify({type: "commonjs"}))
+}
+
+const manifest = getManifest();
+const serverBuildModule = "./.cache/server-build.js";
+const serverBuildEntry = createServerBuildEntry(remixConfig, manifest);
+fs.writeFileSync(serverBuildModule, serverBuildEntry, "utf8");
+
+
+/**
+ * @type {import('webpack').Configuration}
+ */
+const config = {
+  name: "server",
+  mode,
+  devtool: mode === "development" ? false : undefined,
+  target: 'async-node',
+  entry: remixConfig.serverEntryPoint
+    ? path.resolve(remixConfig.rootDirectory, remixConfig.serverEntryPoint)
+    : serverBuildModule,
+  experiments: isModule ? {outputModule: true} : undefined,
+  externalsType: isModule ? "module" : undefined,
+  externalsPresets: {node: true},
+  externals: [
+    nodeExternals({
```

Comment:
disable externals or opt shared modules and their dependencies out of externalization